models/training-tuning-scripts/root-cause-models/sequence_classifier.py

```diff
@@ -0,0 +1,254 @@
+        train_gdf["text"] = train_data
+        train_gdf["label"] = labels
+
+        train_dataset = Dataset(train_gdf)
+        train_dataloader = DataLoader(train_dataset, batchsize=batch_size)
+
+        self._config_optimizer(learning_rate)
+        self._model.train()  # Enable training mode
+        self._tokenizer = SubwordTokenizer(self._hashpath, do_lower_case=True)
+
+        for _ in trange(epochs, desc="Epoch"):
+            tr_loss = 0  # Tracking variables
+            nb_tr_examples, nb_tr_steps = 0, 0
+            for df in train_dataloader.get_chunks():
+                b_input_ids, b_input_mask = self._bert_uncased_tokenize(df["text"], max_seq_len)
+
+                b_labels = torch.tensor(df["label"].to_array())
+                self._optimizer.zero_grad()  # Clear out the gradients
+                # forwardpass
+                loss = self._model(b_input_ids, token_type_ids=None, attention_mask=b_input_mask, labels=b_labels)[0]
+
+                loss.sum().backward()
+                self._optimizer.step()  # update parameters
+                tr_loss += loss.sum().item()  # get a numeric value
+                nb_tr_examples += b_input_ids.size(0)
+                nb_tr_steps += 1
+
+            print("Train loss: {}".format(tr_loss / nb_tr_steps))
+
+    def evaluate_model(self, test_data, labels, max_seq_len=128, batch_size=32):
+        """
+        Evaluate trained model
+
+        :param test_data: test data to evaluate model
+        :type test_data: cudf.Series
+        :param labels: labels for each element in test_data
+        :type labels: cudf.Series
+        :param max_seq_len: Limits the length of the sequence returned by tokenizer. If tokenized sentence is shorter
+            than max_seq_len, output will be padded with 0s. If the tokenized sentence is longer than max_seq_len it
+            will be truncated to max_seq_len.
+        :type max_seq_len: int
+        :param batch_size: batch size
+        :type batch_size: int
+
+        Examples
+        --------
+        >>> from cuml.preprocessing.model_selection import train_test_split
+        >>> emails_train, emails_test, labels_train, labels_test = train_test_split(train_emails_df,
+                                                                                    'label',
+                                                                                    train_size=0.8)
+        >>> sc.evaluate_model(emails_test, labels_test)
+        """
+        self._model.eval()
+        test_gdf = cudf.DataFrame()
+        test_gdf["text"] = test_data
+        test_gdf["label"] = labels
+
+        test_dataset = Dataset(test_gdf)
+        test_dataloader = DataLoader(test_dataset, batchsize=batch_size)
+
+        eval_accuracy = 0
+        nb_eval_steps = 0
+        for df in test_dataloader.get_chunks():
+            b_input_ids, b_input_mask = self._bert_uncased_tokenize(df["text"], max_seq_len)
+            b_labels = torch.tensor(df["label"].to_array())
```

Comment:
to_array() is no longer in cudf 22.08. should to_numpy()

Reply:
updated the scripts